x-pack/plugins/endpoint/server/routes/alerts/list/lib/index.ts

```diff
@@ -36,7 +36,10 @@ export const getRequestData = async (
       : config.alertResultListDefaultDateRange) as unknown) as TimeRange,
 
     // Filtering
-    query: request.query.query,
+    query:
+      request.query.query !== undefined
+        ? ((decode(request.query.query) as unknown) as Query)
```

Comment:
Does `decode` need any error handling or anything? Curious because this will have typescript trust that the value is a Query. What happens if the decode fails?

Reply:
@oatkiller In this case, the schema will have already validated the input, so we should be okay, no? Or do you think we should handle this differently? In any case, we should have API-level tests to check for a `400` and ensure there's no `500` when invalid input is provided. @peluja1012 <<<